merino-web/src/errors.rs

```diff
@@ -18,17 +45,94 @@ pub enum HandlerError {
     MalformedHeader(&'static str),
 }
 
-impl ResponseError for HandlerError {
-    fn status_code(&self) -> StatusCode {
+impl HandlerErrorKind {
+    /// Convert the error to an HTTP status code.
+    pub fn status_code(&self) -> StatusCode {
         match self {
             Self::Internal => StatusCode::INTERNAL_SERVER_ERROR,
             Self::MalformedHeader(_) => StatusCode::BAD_REQUEST,
         }
     }
```

Comment:
Why do you have this defined here in the "bare" impl block, instead of just defining it directly in `impl ResponseError`?